packages/block-editor/src/components/inserter/index.js

```diff
@@ -105,10 +129,47 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
-
+		const { hasInserterItems, hasOneAllowedItem, getOneAllowedItem } = select( 'core/block-editor' );
+		const allowedBlock = getBlockType( getOneAllowedItem( rootClientId ) );
 		return {
 			hasItems: hasInserterItems( rootClientId ),
+			hasOneAllowedItem: hasOneAllowedItem( rootClientId ),
+			blockTitle: allowedBlock ? allowedBlock.title : '',
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, { select } ) => {
+		// eslint-disable-next-line no-restricted-syntax
```

Comment:
We require each `eslint` disable rule to be justified with a comment. Why disable here?

Reply:
The reason was that there is a rule for no logic in withDispatch but since there is only one thing returned there is no need to extract this so I returned in the object body instead.